core/arch/arch_exports.h

```diff
@@ -327,9 +327,13 @@ emit_detach_callback_final_jmp(dcontext_t *dcontext,
 /* note that the microsoft compiler will not enregister variables across asm
  * blocks that touch those registers, so don't need to worry about clobbering
  * eax and ebx */
+#    define ATOMIC_1BYTE_READ(addr_src, addr_res)               \
+        do {                                                    \
+            *(BYTE *)(addr_res) = *(volatile BYTE *)(addr_src); \
```

Comment:
Why are there two strategies of atomic reads: the asm block and the volatile pointer deref?

Reply:
Oh, the separate Windows and Linux ones for x86?  The Visual Studio 64-bit compiler does not support *any* inline assembly is the reason.  I suppose we could have the C code for Linux too?  I was just following the existing split.  Many of the more complex operations can't be written in C so only a few could be shared like that (they use special intrinsics for Visual Studio).  I guess I would lean toward just leaving it unless you feel strongly?

Reply:
I was just curious about the difference. We can continue following the existing split.